config/config.go

```diff
@@ -38,6 +38,7 @@ type Counter struct {
 	Name           string        `yaml:"name"`
 	Help           string        `yaml:"help"`
 	PerfEventArray bool          `yaml:"perf_event_array"`
+	RingBuf        bool          `yaml:"ring_buf"`
 	FlushInterval  time.Duration `yaml:"flush_interval"`
```

Comment:
In fact, I think this may not be a good idea to create another option alone with  PerfEventArray, as more map types may be introduced in the future.

I think there might be two ways to handle this issue better:

1. we can infer the map type directly from the bpf object, with the given map name. So we may remove this field in the config struct.
2. Add an enum type to represent map type.

But they may bring some API break changes, so I left it here...Which do you like?

Reply:
I think we should infer the type, but not from the name. Every map has a type:

* https://pkg.go.dev/github.com/aquasecurity/libbpfgo@v0.4.2-libbpf-1.0.1#BPFMap.Type

And we can check `MapType` for what we want:

* https://pkg.go.dev/github.com/aquasecurity/libbpfgo@v0.4.2-libbpf-1.0.1#MapType

There's `MapTypePerfEventArray ` and `MapTypeRingbuf` there already.